FirebaseCore/Tests/Unit/FIRAppTest.m

```diff
@@ -50,6 +83,20 @@ + (nullable NSNumber *)readDataCollectionSwitchFromUserDefaultsForApp:(FIRApp *)
 
 @end
 
+static FIRHeartbeatLoggerFake *heartbeatLoggerFake;
+
+@implementation FIRApp (Internal)
+
++ (id<FIRHeartbeatLoggerProtocol> (^)(NSString *appID))heartbeatLoggerProviderForAppID {
+  return ^id<FIRHeartbeatLoggerProtocol>(NSString *_) {
+    return heartbeatLoggerFake;
+  };
+}
+
+@end
```

Comment:
I think an improvement here would be to expose a setter for FIRApp's `heartbeatLoggerProviderForAppID` and then set it to return a fake one in the `-[FIRAppTest setUp]`

Reply:
Discussed approach with Ryan offline.